TrafficSimulator/Assets/Car/CarSpawner.cs

```diff
@@ -0,0 +1,241 @@
+                }
+            }
+        }
+
+        private void AddLanesToList()
+        {
+            // Loop through all roads
+            for (int i = 0; i < _roadSystem.RoadCount; i++)
+            {
+                _roads[i].OnChange();
+
+                // Loop through all lanes
+                for (int j = 0; j < _roads[i].LaneCount; j++)
+                {
+                    _lanes.Add(_roads[i].Lanes[j]);
+                }
+            }
+        }
+
+        private void CalculateLaneLengths()
+        {
+            LaneNode _laneNodeTemp;
+            float laneLength;
+
+            // Loop through all lanes
+            for (int j = 0; j < _lanes.Count; j++)
+            {
+                laneLength = 0;
+                _laneNodeTemp = _lanes[j].StartNode;
+
+                // Loop through all lane nodes
+                for (int k = 0; k < _lanes[j].StartNode.Count - 1; k++)
+                {
+                    // Calculate the length of the lane
+                    laneLength = laneLength + Vector3.Distance(_laneNodeTemp.Position, _laneNodeTemp.Next.Position);
+                    _laneNodeTemp = _laneNodeTemp.Next;
+                }
+                _lengths.Add(laneLength);
+            }
+        }
+
+        private void CalculateLaneRatios()
+        {
+            float totalLength = 0;
+
+            // Loop through all lane lengths
+            for (int i = 0; i < _lengths.Count; i++)
+            {
+                totalLength = totalLength + _lengths[i];
+            }
+
+            // Loop through all lane lengths
+            for (int i = 0; i < _lengths.Count; i++)
+            {
+                _ratios.Add(_lengths[i] / totalLength);
+            }
+        }
+
+        private void CalculateLaneIndexes()
+        {
+            int laneIndex;
+
+            // Loop through all roads
+            for (int i = 0; i < _roadSystem.RoadCount; i++)
+            {
+                _roads[i].OnChange();
+                laneIndex = 0;
+
+                // Loop through all lanes
+                for (int j = 0; j < _roads[i].LaneCount; j++)
+                {
+                    _indexes.Add(laneIndex);
+                    laneIndex = laneIndex + 1;
+                }
+            }
+        }
+
+        private void CalculateMaxCarsForLanes()
+        {
+            // Loop through all lanes
+            for (int j = 0; j < _lanes.Count; j++)
+            {
+                // Calculate the max cars for the lane
+                _maxCarsPerLane.Add((int)(_lengths[j] / CarLength));
+            }
+        }
+
+        private void SpawnCarsWithPercentage(float MaxCarsPercentage) 
+        {
+            for (int i = 0; i < _lanes.Count; i++)
+            {
+                // Calculate the number of cars to spawn
+                int carsToSpawn = Mathf.CeilToInt(_maxCarsPerLane[i] * MaxCarsPercentage);
+                if (carsToSpawn == 0)
```

Comment:
Add a comment that this is to avoid division by zero for transparency and maintainability